docs/src/main/asciidoc/chapters/iterator_design.txt

```diff
@@ -182,6 +182,12 @@ early programming assignments which implement their own tree data structures. `d
 copy on its sources (the children), copies itself, attaches the copies of the children, and
 then returns itself.
 
+==== `close`
+
+This method should close any resources created by the iterator.  In particular, this method should call `close`
+on the `source` provided by the first argument of the `init` method.  By default this method will do nothing.
```

Comment:
Maybe expand this to also specifically call out SKVI's which are created via deepCopy? Something like: ".. of the `init` method[ as well as any other Iterators created via invoking `deepCopy()`]"? Phrasing could use work :)

What about the guarantees on `close()` being called? I re-read Dylan's earlier comment -- something like "`close()` will be invoked in all scenarios except those resulting in JVM failure such as an `OutOfMemoryError`"?